### PR TITLE
Removing unused code, take 2.

### PR DESCRIPTION
**Pull request by paulp**
Incorporating knowledge of what failed on the first attempt. The list of failures probably looked like a lot more than it was; only five methods were implicated, but some were called a lot. This PR restores those methods and marks them as externally used until we have a more permanent mechanism for assuring their survival.

Review by @dotta (confirmation that everything builds against this would be very helpful.)

### CODE REVIEW

File: src/compiler/scala/reflect/reify/codegen/GenUtils.scala [hunk 1]
```diff
@@ -90,22 +90,6 @@ trait GenUtils {
   /** An (unreified) path that refers to term definition with given fully qualified name */
   def termPath(fullname: String): Tree = path(fullname, newTermName)
 
-  /** An (unreified) path that refers to type definition with given fully qualified name */
-  def typePath(fullname: String): Tree = path(fullname, newTypeName)
```
Comment (soc) [inline src/compiler/scala/reflect/reify/codegen/GenUtils.scala:94]: Is this intentional or just an oversight?

Comment (paulp) [inline src/compiler/scala/reflect/reify/codegen/GenUtils.scala:94]: "Is that a donkey or a baseball?"

**Comment by dotta** [comment]
Paul, two Sbt modules that we bundle in the IDE for incremental compilation do not compile anymore. Sorry for the trouble, yesterday I must have messed up something when locally publishing the Scala artifacts.

Here is the error (it is the same for both Sbt modules: _compile_ and _compiler-interface_):

> [error] /Users/mirco/Projects/ide/uber-build/xsbt/compile/Eval.scala:119: value scalaAnyRefConstr is not a member of object Eval.this.global.gen
> [error]       def moduleBody = Template(List(gen.scalaAnyRefConstr), emptyValDef, List(emptyInit, method))
> [error]                                                                       ^
> [error] one error found

Then, while scala-refactoring now compiles fine, there are a number of failing tests (which runs just fine with the last deployed 2.11.0-SNAPSHOT). Maybe we should ask for Mirko (@misto) feedback.

> Failed tests: 
>   findReferencesToLocal(scala.tools.refactoring.tests.analysis.DeclarationIndexTest)
>   findClassDeclarationToMethodParameter(scala.tools.refactoring.tests.analysis.DeclarationIndexTest)
>   findClassDeclarationFromMethodParameter(scala.tools.refactoring.tests.analysis.DeclarationIndexTest)
>   referencesToLazyVal211(scala.tools.refactoring.tests.analysis.DeclarationIndexTest)
>   referencesToTypesInAppliedTypes(scala.tools.refactoring.tests.analysis.DeclarationIndexTest)
>   findReferencesToMethod(scala.tools.refactoring.tests.analysis.MultipleFilesIndexTest)
>   findReferencesToClass(scala.tools.refactoring.tests.analysis.MultipleFilesIndexTest)
>   findReferencesToTraitMethod(scala.tools.refactoring.tests.analysis.MultipleFilesIndexTest)
>   findInImports(scala.tools.refactoring.tests.analysis.MultipleFilesIndexTest)
>   overriddenMethods(scala.tools.refactoring.tests.analysis.MultipleFilesIndexTest)
>   overriddenMethods2(scala.tools.refactoring.tests.analysis.MultipleFilesIndexTest)
>   overriddenMethods3(scala.tools.refactoring.tests.analysis.MultipleFilesIndexTest)
>   overriddenMethods4(scala.tools.refactoring.tests.analysis.MultipleFilesIndexTest)
>   methodNameCollision(scala.tools.refactoring.tests.analysis.NameValidationTest)
>   simpleUnusedType(scala.tools.refactoring.tests.implementations.imports.UnusedImportsFinderTest)
>   typeIsUsedAsVal(scala.tools.refactoring.tests.implementations.imports.UnusedImportsFinderTest)
>   typeIsImportedFrom(scala.tools.refactoring.tests.implementations.imports.UnusedImportsFinderTest)
>   wildcardImports(scala.tools.refactoring.tests.implementations.imports.UnusedImportsFinderTest)
> wildcardImportsFromValsAreIgnored(scala.tools.refactoring.tests.implementations.imports.UnusedImportsFinderTest)
>   markImportedOccurrences(scala.tools.refactoring.tests.implementations.MarkOccurrencesTest)
>   markImportedAndRenamedOccurrences(scala.tools.refactoring.tests.implementations.MarkOccurrencesTest)
> markImportedAndRenamedOccurrencesRenamedSelected(scala.tools.refactoring.tests.implementations.MarkOccurrencesTest)
> 
> Tests run: 874, Failures: 22, Errors: 0, Skipped: 15

Finally, there is a new error in the IDE side:

> [ERROR] /Users/mirco/Projects/ide/uber-build/scala-ide/org.scala-ide.sdt.debug.tests/src/scala/tools/eclipse/debug/classfile/ClassfileParser.scala:76: error: value nextBytes is not a member of scala.tools.nsc.symtab.classfile.AbstractFileReader
> [ERROR]         bytecode = reader.nextBytes(codeLen)

**Comment by misto** [comment]
@dotta can you send me the detailed error reports of the failures?

**Comment by dotta** [comment]
Sure, here you go http://dl.dropbox.com/u/50945416/scala-refactoring-surefire-reports.zip

**Comment by misto** [comment]
Ok, it seems all the failures have the same cause: "Race condition detected: You are running a presentation compiler method outside the PC thread." Should be easy to fix!

**Comment by dotta** [comment]
Oh, I see, thanks Mirko for looking into it! And, I just noticed that our nightly for scala-refactoring against 2.11.0-SNAPSHOT is not running tests. That explains why I did not see the failing tests before.

Bottom line, I believe we can ignore the failing tests in scala-refactoring, as they are not related to this PR. If we can sort out the two remaining compilation errors, you'll have my :+1: ;-)

**Comment by paulp** [comment]
If I understood which errors were legit and which weren't, the push I just did should get everything compiling.

**Comment by paulp** [comment]
PLS REBUILD ALL

**Comment by dotta** [comment]
> If I understood which errors were legit and which weren't, the push I just did should get everything compiling.

Yes, now it's all good from the IDE side.

**Comment by adriaanm** [comment]
sure sounds like an LGTM to me! excellent work, gang!

**Comment by adriaanm** [comment]
so long, lines of code!

**Comment by retronym** [comment]
Surely as a reward @paulp gets the honour of driving the stake through patmat classico on master.

**Comment by adriaanm** [comment]
that, and a complimentary cappuccino

File: src/compiler/scala/reflect/reify/codegen/GenUtils.scala (hunk 1 above)
Comment (soc) [inline src/compiler/scala/reflect/reify/codegen/GenUtils.scala:94]: :-)

**Pull request merged**